pymatgen/analysis/ewald.py

```diff
@@ -619,7 +619,7 @@ def best_case(self, matrix, m_list, indices_left):
         interaction_correction = np.sum(step3)
 
         if self._algo == self.ALGO_TIME_LIMIT:
-            elapsed_time = datetime.datetime.now(datetime.timezone.utc) - self._start_time
+            elapsed_time = datetime.now(timezone.utc) - self._start_time
```

Comment:
This change need to be reverted unless the following issue can be resolved:
```python
import datetime

start_time_old = datetime.datetime.utcnow()  # 2024-03-25 08:46:23.748342 

start_time_new = datetime.datetime.now(datetime.UTC)  # 2024-03-25 08:46:23.748472+00:00

print(start_time_old, start_time_new)  


dt = datetime.datetime.strptime(
    str(start_time_old), "%Y-%m-%d %H:%M:%S.%f"
)
print(dt)


dt = datetime.datetime.strptime(
    str(start_time_new), "%Y-%m-%d %H:%M:%S.%f"
)
print(dt)
```

Which raises:
```python
Traceback (most recent call last):
  File "/Users/yang/developer/test/test.py", line 16, in <module>
    dt = datetime.datetime.strptime(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/_strptime.py", line 554, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.2_1/Frameworks/Python.framework/Versions/3.12/lib/python3.12/_strptime.py", line 336, in _strptime
    raise ValueError("unconverted data remains: %s" %
ValueError: unconverted data remains: +00:00
```

Reply:
if it's just there to measure elapsed time then it's not using the right API anyway and needlessly complicated as a result. have a look at [`time.perf_counter`](https://docs.python.org/3/library/time.html#time.perf_counter) instead

Reply:
Thanks I would have a closer look, and perhaps report this to `Python` repo, which is quite similar to [this](https://github.com/python/cpython/issues/109964). 

It should have been a simple deprecation replacement (`datetime.datetime.utcnow()` -> `datetime.datetime.now(datetime.UTC)`). But for some reason these two doesn't return the exact same value (with an additional `+00:00` at the end from the new API), which leads to format error.

Reply:
I decide to revert this deprecation change for now until I could figure out a way to avoid `datetime.datetime.now(datetime.UTC)` from output timezone info.